### DISCUSSION TITLE
Gastro Bill does not print to the Kitchen Printer instead prints to the default printer when using Nexo Printer Server

### DISCUSSION THREAD

**Discussion by mbugikp** [Blair2004/NexoPOS#479] [Ideas]
**Describe the bug**
 Gastro Bill does not print to the Kitchen Printer instead prints to the default printer when using Nexo Printer Server

**To Reproduce**
I am using Nexo 4.6.8 and Gastro 4.6.2 however, i am not able to use NPS with Gastro for KOT (Kitchen Order) and billing.
My Scenario is as follows i have Bar printer, Restaurant printer and Garden Printer, When waiter makes an order
it goes to the Restaurant Printer(Restaurant Kitchen), or Bar Printer(Bar Kitchen), or Garden Printer( Garden Kitchen),
and it gets printed but, the waiter then has to go to orders  
![orders1](https://user-images.githubusercontent.com/13999977/147812489-5a71a7f9-ac11-4271-a2ef-25b978c0e738.png)
and then clicks print on the held order. 
![print order](https://user-images.githubusercontent.com/13999977/147812523-fdece76e-0947-4af0-88b7-794b4fa9036c.png)

This is where it gets tricky if i have the printing gateway set as NPS (POS Settings) it will print to the default printer, now if the 
waiter is operating from the bar and the default printer is set to the Restaurant Printer the waiter has to go to the
Restaurant Printer to get the printed bill this is time wasting and not practical, so i simply set web as the default
printing gateway.
![Printing Gateway](https://user-images.githubusercontent.com/13999977/147812578-3b7f0205-9b7a-492e-9081-2cae708fda8f.png)

One solution to this issue is to have the bill printed together with the order that is the kitchen order gets printed,
then the bill follows it automatically.
Without this i cannot also use a mobile device or tablet for the simple reason that it will print to the default printer

**Expected behavior**
One solution to this issue is to have the bill printed together with the order that is the kitchen order gets printed,
then the bill follows it automatically.

**Screenshots**

**Environment:**
 - NexoPOS 4.6.8 
 - Premium Extensions Nexo Print Adapter (4.5.0), Gastro (4.6.2)

**Comment by Blair2004** [comment]
Hi, 
Understood, let me try that.

**Comment by mbugikp** [comment]
I will appreciate that.
Regards

**Comment by Blair2004** [comment]
Just to understand better. 
So you're expecting that from the orders popup, you can print to the kitchen ?
What do you mean by default printer ?

**Comment by mbugikp** [comment]
What do you mean by order pop-up? When I make an order it goes to the
kitchen and gets printed automatically that print out is called KOT for
kitchen order ticket I hope we are together so far, now after that the
waiter needs a bill to take to the customer, this bill is what I am saying
should be printed after the order is placed or immediately the order is
placed, or should accompany the order, I know there are scenarios where the
bill is not printed immediately after the order so we can have a setting
where we activate that option or disable it.



Regards

**Comment by Blair2004** [comment]
Ok, we'll be performing some test to see if the complete printing process works well.

**Comment by mbugikp** [comment]
Default printer

**Comment by Blair2004** [comment]
I'm not sure what you mean by default printer Can you provide any screenshot ?

**Comment by magedfathi** [comment]
@Blair2004  the default printer is the one printing the customer recipt

**Comment by Blair2004** [comment]
The improvement suggested will cause for some a paper waste. Instead, from the order list, we might instead display a list of available printer. We'll hold is as an improvement.

**Reply by mbugikp** [reply]
![Print bill with order](https://user-images.githubusercontent.com/13999977/149660563-7c1c32d1-b89f-4d1f-8033-ae4674bd7b02.png)

Hi @Blair2004 , my suggestion is to add an option to enable or disable this feature as shown above, this will save time for the waiter does not have to search and then print the order

**Reply by Blair2004** [reply]
If the issue is to not be able to have the receipt printed at the specific printer, I believe it's handy to provide an option to choose the printer.
When the order is placed, according to the settings, you can yet have the receipt printed if the order is unpaid, or paid. That new option is conflicting with "Printing Enabled For".

**Reply by mbugikp** [reply]
@Blair2004  The solution we are looking for should make selling faster not adding many steps before printing a bill, we should therefore command the system to print and a bill to a pre-defined printer(s) just like orders are sent to the kitchen printer send the bills to kitchen printer, bar printer, restaurant printer, etc, but if we have a system where the waiter will choose a printer every time he wants to print a bill which in this case is every time and order is placed we shall be adding too many steps for the waiter before making a sale which is time consuming, time which the waiter does not have in a busy restaurant.

Note currently we cannot use the system with tablet or mobile device because as mentioned earlier the system will print to the default printer and we have three printers thus i could be in the garden restaurant i make an order the order is sent to the kitchen printer correctly but when i print the bill it gets printed in the bar(default) printer, now i have to walk to the bar about 50m just to pick the bill and then come back (50m) to the garden restaurant remember we have a printer in the garden restaurant. We therefore do not use the nexo print server and we do not use mobile device or tablet.

Check let us know.

**Reply by Blair2004** [reply]
What about using the cash register so you can define from which printer every receipt should be printed ? Normally, when the cash register is enabled and a sale is made, the printer assigned to that register must be used.

A cash register doesn't absolutely need to be a real cash register, you can assume it to be a terminal where sales are made and receipt printed.

I think in that case it should be a better solution, as the printer will always be predefined.

Let me know.

**Reply by mbugikp** [reply]
@Blair2004 
Thank you, let me try that tomorrow.

**Comment by mbugikp** [comment]
I have used the cash register option, the kitchen order gets printed but system shows me an error
![Kitchen Order gets printed but i am getting this error](https://user-images.githubusercontent.com/13999977/149706891-38e7da67-4bc7-4c9f-bdc4-4b40281a7e00.png)


When i try to print the bill i get another error
![unable to print order with cash register on and nexo print server](https://user-images.githubusercontent.com/13999977/149706993-02534486-b012-408e-9fb2-fb500a325f8d.png)

This is my current configuration on the cash register
![Cash Register Printer](https://user-images.githubusercontent.com/13999977/149707073-866a19f0-1c4d-451c-94db-681ccb77d246.png)

**Reply by Blair2004** [reply]
Try to update to 4.6.10

**Comment by mbugikp** [comment]
@Blair2004 your suggestion 
" **_The improvement suggested will cause for some a paper waste. Instead, from the order list, we might instead display a list of available printer. We'll hold is as an improvement._** "

was most welcome. 
considering this was Jan 11th 2022 we should revisit this issue. check if we can display a list of available printers.

Regards
Mbugi
